src/interfaces/node.cpp

```diff
@@ -259,12 +259,12 @@ class NodeImpl : public Node
     }
     std::unique_ptr<Wallet> loadWallet(const std::string& name, std::string& error, std::string& warning) override
     {
-        return MakeWallet(LoadWallet(*m_interfaces.chain, name, error, warning));
+        return MakeWallet(LoadWallet(*m_node.chain, name, error, warning));
     }
     WalletCreationStatus createWallet(const SecureString& passphrase, uint64_t wallet_creation_flags, const std::string& name, std::string& error, std::string& warning, std::unique_ptr<Wallet>& result) override
     {
         std::shared_ptr<CWallet> wallet;
-        WalletCreationStatus status = CreateWallet(*m_interfaces.chain, passphrase, wallet_creation_flags, name, error, warning, wallet);
+        WalletCreationStatus status = CreateWallet(*m_node.chain, passphrase, wallet_creation_flags, name, error, warning, wallet);
```

Comment:
Do `Node` really need a reference to `Chain`, `ChainImpl` has already a reference to `Node` that's a bit confusing ? I mean `createWallet` and `loadWallet` couldn't be method of `Wallet` interface instead ? Would it work for the GUI ?

More generally, if we refactor RPC code to encapsulate in a `RPCServer` class, I see a `WalletRPC` subclass with a `Chain` member and a `NodeRPC` subclass with a `Node` member. And `Node` maybe initialize differently to avoid access to uneeded state which would address some concerns of https://github.com/bitcoin/bitcoin/pull/16839#discussion_r325866327

Reply:
Thanks for answers (sorry for delay), wasn't implying to do changes in this PR. Apart of RPC framework which need design goals, I think other one would make code nicer in future PRs, bookmarked.